utest/src-0/utest/Tests.scala

```diff
@@ -0,0 +1,35 @@
+package utest
+
+import scala.quoted.{ given, _ }
+import scala.tasty._
+
+import utest.framework.{TestCallTree, Tree => UTree }
+import scala.collection.mutable
+
+/**
+  * Represents a single hierarchy of tests, arranged in a tree structure, with
+  * every node having a name and an associated executable test.
+  *
+  * The two hierarchies are parallel: thus you can inspect the `nameTree` to
+  * browse the test listing without running anything, and once you decide which
+  * test to run you can feed the `List[Int]` path of that test in the `nameTree`
+  * into the `callTree` to execute it and return the result.
+  */
+case class Tests(nameTree: UTree[String], callTree: TestCallTree)
+object Tests{
+  inline def apply(expr: => Unit): Tests = ${testsImpl('expr)}
```

Comment:
Could we move the `def apply` and macro-impl to version specific code? I know this file is pretty small so it's not a *lot* of copy paste, but keeping the copy-paste to a minimum helps make the version-specific parts of the codebase immediately clear upon first inspection, whereas right now I have to diff the two files in my head to see what's important.

Even if we have to pay some boilerplate extending traits etc. to make that happen, I think it's still worth it

Reply:
Do you mean making the `case class Tests` shared and keep the rest version specific? I don't think the companion objects can be shared since they are all about macros.

Reply:
I mean keep the `case class` and companion `object` shared, but have the companion extend a small version-specific `trait` that provides the macro definition and move the macro definition onto the `trait`